bundle/config/mutator/translate_paths.go

```diff
@@ -94,14 +103,16 @@ func (m *translatePaths) translateNotebookPath(literal, localPath, remotePath st
 }
 
 func (m *translatePaths) translateFilePath(literal, localPath, remotePath string) (string, error) {
-	_, err := os.Stat(localPath)
+	nb, _, err := notebook.Detect(localPath)
 	if os.IsNotExist(err) {
 		return "", fmt.Errorf("file %s not found", literal)
 	}
 	if err != nil {
-		return "", fmt.Errorf("unable to access %s: %w", localPath, err)
+		return "", fmt.Errorf("unable to determine if %s is not a notebook: %w", localPath, err)
+	}
+	if nb {
+		return "", ErrIsNotebook{localPath}
```

Comment:
This introduces an asymmetry between notebook handling and file handling. Could you apply the same pattern to the notebook detection and return `ErrIsNotNotebook` (or whatever) and produce custom errors in the translation code below?